src/main/java/com/hivemq/cli/commands/options/SubscribeOptions.java

```diff
@@ -48,37 +55,164 @@ public class SubscribeOptions {
                         description = "Quality of service for the corresponding topics (default for all: 2)")
     private @NotNull MqttQos @NotNull [] qos;
 
+    private @NotNull Mqtt5UserProperty @Nullable [] userProperties = null;
+    private boolean userPropertiesFromOption = false;
+    private boolean userPropertiesFromLegacyOption = false;
+
     @SuppressWarnings("unused")
-    @CommandLine.Option(names = {"-up", "--userProperty"},
+    @CommandLine.Option(names = {"--user-property"},
                         converter = Mqtt5UserPropertyConverter.class,
                         description = "A user property of the subscribe message")
-    private @Nullable Mqtt5UserProperty @Nullable [] userProperties;
+    private void userProperties(final @NotNull Mqtt5UserProperty @NotNull [] userProperties) {
+        if (userPropertiesFromLegacyOption) {
```

Comment:
I feel that this log is redundant, but I'm ok with leaving it for emphasis

Reply:
It is actually an exception so that we create mutual exclusion between both of the user properties. This has multiple reasons:
- We cannot guarantee user property ordering with two different options as pico-cli only gives us executions with [first-el], [fist-el, second-el], ... As ordering is important in the MQTT spec we would break it.
- User properties can occur multiple time. I do not think someone would use the legacy option and the current one to write different user properties with the same name and value. Feels like it is a bug then.
- ArgGroups are currently broken in the current PicoCLI version if combined with mixins..